src/pruning_scheduler.py

```diff
@@ -16,13 +13,17 @@ def should_prune(self):
 
 
 class PruneAfterEpochs(PruningScheduler):
-    epochs_since_pruning = 0
+    def __init__(self, epochs_before_pruning, pruning_interval):
+        self.epochs_before_pruning = epochs_before_pruning
+        self.pruning_interval = pruning_interval
+        self.epochs_since_pruning = 0
 
     def should_prune(self):
-        if self.epochs_before_pruning is None:
-            raise AttributeError("Epochs before pruning not set")
+        if self.epochs_before_pruning < 0:
```

Comment:
I do not get this condition. Is it supposed to be:
```suggestion
        if self.epochs_before_pruning > 0:
```